code/datums/components/obj_projectile_damage.dm

```diff
@@ -14,9 +14,20 @@ TYPEINFO(/datum/component/obj_projectile_damage)
 /datum/component/obj_projectile_damage/proc/projectile_collide(owner, var/obj/projectile/P)
 	var/obj/O = parent
 	if(P.proj_data.damage_type & (D_KINETIC | D_ENERGY | D_SLASHING))
+		var/gib_type = /obj/decal/cleanable/machine_debris
+		var/do_gib = TRUE
+		var/do_streak = TRUE
+		if (istype(O, /obj/structure/woodwall))
```

Comment:
Ah no, this is bad. The calling object should provide the type of gib to spawn as a parameter to the component (defaulting to what it was before). Components should generally be agnostic to what type of thing they are attached to (otherwise they're just more tightly coupled handling that could be implemented on the object itself)